packet-solutions/xdp_prog_kern_02.c

```diff
@@ -46,17 +55,33 @@ int xdp_patch_ports_func(struct xdp_md *ctx)
 	}
 
 	if (ip_type == IPPROTO_UDP) {
+		//struct udphdr udphdr_old;
+		//__u32 csum;
 		if (parse_udphdr(&nh, data_end, &udphdr) < 0) {
 			action = XDP_ABORTED;
 			goto out;
 		}
+		/*csum = udphdr->check;
+		*udphdr_old = *udphdr;
+		*udphdr->dest = bpf_htons(bpf_ntohs(udphdr->dest) - 1);
+
+		*csum = bpf_csum_diff((__be32 *)&udphdr_old, 4, (__be32 *)udphdr, 4, ~csum);
+		*udphdr->check = csum_fold_helper(csum);
+		*/
```

Comment:
If we're keeping the full code I think it should be in an explanatory comment above the other code, not just commented-out with no explanation like this. So maybe something like:
```C
/*
 * We need to update the packet checksum when modifying the header.
 * RFC1071 contains an algorithm for in-place updating, which is what we use here
 * since we're always just decrementing the port number. Another option would be
 * to recompute the full checksum, like:
 *
 * struct udphdr udphdr_old = *udphdr;
 * __u32 csum = udphdr->check
 *
 * udphdr->dest = bpf_htons(bpf_ntohs(udphdr->dest) - 1);
 * csum = bpf_csum_diff((__be32 *)&udphdr_old, 4, (__be32 *)udphdr, 4, ~csum);
 * udphdr->check = csum_fold_helper(csum);
 */
```

Reply:
@tohojo I sent you a  email I got from your github profile. Could you reply? thanks.